connector/src/main/java/org/geysermc/connector/network/translators/collision/CollisionManager.java

```diff
@@ -254,10 +238,8 @@ public boolean correctPlayerPosition() {
         touchingScaffolding = false;
         onScaffolding = false;
 
-        List<Vector3i> collidableBlocks = getPlayerCollidableBlocks();
-
         // Used when correction code needs to be run before the main correction
-        for (Vector3i blockPos : collidableBlocks) {
+        for (Vector3i blockPos : playerCollidableBlocksIterator()) {
```

Comment:
So we're creating the same iterator twice here, and that means we're creating twice as many Vector3i classes as we were before. Honestly this code was already interesting because we're also calling `getCollisionAt` twice in a row. But, if a way around this current regression could be found, that would be nice.

Alternatively we could create `nextX()`, `nextY()` and `nextZ()` that all return primitives? That would probably require a good refactor, though.

Reply:
That is also true for `computeCollisionOffset` which gets called ~12x.
I did the refactor at https://github.com/davchoo/Geyser/commit/8a3e95924a6b8572322b228c73807af42807147f since I was curious as to how it would turn out. Its not exactly pretty, but it works.
As for `getCollisonAt` being called more than once, I guess the only way around it is to go back to creating ArrayLists. 

I think I'll revert the iterator changes and try caching the blocks.

Reply:
I like those changes a lot, especially if it means we can get rid of those ThreadLocal variables.